common/lib/xmodule/xmodule/fields.py

```diff
@@ -249,3 +249,25 @@ def enforce_type(self, value):
             return value
 
         return self.from_json(value)
+
+
+class WarningFieldSingleValueMixin(object):
+    """
+    Wraps around the actual value of another xmodule.field.Field.
+    If the value of this field is set on the front-end, a warning
+    from the backend is shown
+    """
+    def __init__(self, warning=None, warning_condition=None, **kwargs):
+        super(WarningFieldSingleValueMixin, self).__init__(**kwargs)
+        self._values = {
```

Comment:
Yes, it's non obvious from this code. I should document the front-end behavior here. The magic happens in `x_module.py:_create_metadata_editor_info`